circuit/src/builder/compiler/expression_lowerer.rs

```diff
@@ -179,18 +189,58 @@ where
             }
         }
 
-        // Pass C: emit arithmetic ops in creation order; tie outputs to class slot if connected
+        // Pass C: collect witness hints and emit `Unconstrained` operatios
```

Comment:
Question: How does ordering work if, for instance, I have some `WitnessHint` whose inputs depend on an arithmetic op, as these are handled _after_ the witness pass? Before this couldn't happen because we were handling all primitive ops (apart from Constant / PI) together, but now I think we may throw a `MissingExprMapping` issue in some cases?

Reply:
You're right. I need to move the push of `Op::Unconstrained` after Pass D

Reply:
Now should be fixed